### PR TITLE
Milestone3

### PR DESCRIPTION
**Pull request by VinylKnight**
Mostly Code.

### CODE REVIEW

File: Assets/Scripts/Door.cs [hunk 1]
```diff
@@ -25,11 +43,25 @@ protected override void Awake()
...
+
+            }
+            else
+            {
+                audioSource.clip = isLockedAudioClip;
+            }
+            base.InteractWith();
+        }
+        
+
```
Comment (BagboBillbins) [inline Assets/Scripts/Door.cs]: tooo many spacesss

**Review by BagboBillbins — APPROVED** [review]

**Comment by BagboBillbins** [comment]
you best get rid of them spaces by the end of this project

File: Assets/Scripts/DetectLookedAtInteractive.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    [SerializeField]
+    private float maxRange = 5.0f;
+
+    /// <summary>
+    /// Event raised when the player looks at a different IInteractive.
+    /// </summary>
+
+    public static event Action<IInteractive> LookedAtInteractiveChanged;
+   
+
```
Comment (dantogno) [inline Assets/Scripts/DetectLookedAtInteractive.cs:25]: remove extra line space here

File: Assets/Scripts/DetectLookedAtInteractive.cs [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        get { return lookedAtInteractive; }
+        private set
+        {
+            bool isInteractiveChanged = value != lookedAtInteractive;
+            if (isInteractiveChanged)
+            {
+                lookedAtInteractive = value;
+                LookedAtInteractiveChanged?.Invoke(LookedAtInteractive);
+            }
+
```
Comment (dantogno) [inline Assets/Scripts/DetectLookedAtInteractive.cs:39]: Remove extra lines / whitespace unless you are systematically adding extra whitespace / line breaks to enhance the readability of your document (for example, by having a blank line separating distinct elements).

File: Assets/Scripts/Door.cs [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        displayText = nameof(Door);
+    }
+    protected override void Awake()
+    {
+        base.Awake();
+        animator = GetComponent<Animator>();
+    }
+    public override void InteractWith()
+    {
+
```
Comment (dantogno) [inline Assets/Scripts/Door.cs:45]: Remove extra lines / whitespace unless you are systematically adding extra whitespace / line breaks to enhance the readability of your document (for example, by having a blank line separating distinct elements).

File: Assets/Scripts/InteractiveObject.cs [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    [SerializeField]
+    protected string displayText = nameof(InteractiveObject);
+    
+    public virtual string DisplayText => displayText;
+    protected AudioSource audioSource;
+
+    protected virtual void Awake()
+    {
+        audioSource = GetComponent<AudioSource>();
+
```
Comment (dantogno) [inline Assets/Scripts/InteractiveObject.cs:17]: extra lines

**Pull request merged**